Indexing fails for Empty Date Field

We are seeing multiple failures in indexing datasets in prod (15K or more) One type of failure occurs due to a subscript out of range error for date fields. 

Caused by: java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
        at java.util.ArrayList.rangeCheck(ArrayList.java:657)
        at java.util.ArrayList.get(ArrayList.java:433)
        at edu.harvard.iq.dataverse.search.IndexServiceBean.addOrUpdateDataset(IndexServiceBean.java:778)
        at edu.harvard.iq.dataverse.search.IndexServiceBean.indexDataset(IndexServiceBean.java:593)
        at edu.harvard.iq.dataverse.search.IndexServiceBean.indexDatasetInNewTransaction(IndexServiceBean.java:286)

--- a/src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java
+++ b/src/main/java/edu/harvard/iq/dataverse/search/IndexServiceBean.java
@@ -775,7 +775,10 @@ public class IndexServiceBean {
                         // no-op. we want to keep email address out of Solr per
                         // https://github.com/IQSS/dataverse/issues/759
                     } else if (dsfType.getSolrField().getSolrType().equals(SolrField.SolrType.DATE)) {
-                        String dateAsString = dsf.getValues_nondisplay().get(0);
+                        String dateAsString = "";
+                        if (!dsf.getValues_nondisplay().isEmpty()) {
+                            dateAsString = dsf.getValues_nondisplay().get(0);
+                        }                      
                         logger.fine("date as string: " + dateAsString);
                         if (dateAsString != null && !dateAsString.isEmpty()) {
                             SimpleDateFormat inputDateyyyy = new SimpleDateFormat("yyyy", Locale.ENGLISH);

